libsel4/include/api/syscall.xsd

```diff
@@ -35,11 +35,34 @@
 
     <xsd:complexType name="ConfigType" >
         <xsd:sequence>
+            <xsd:element name="condition" minOccurs="0" maxOccurs="1" type="Unary"/>
             <xsd:element name="syscall" minOccurs="1" maxOccurs="unbounded" type="SyscallType"/>
         </xsd:sequence>
-        <xsd:attribute name="condition" type="xsd:string" />
     </xsd:complexType>
 
+    <xsd:group name="ConfigGroup">
+        <xsd:choice>
+            <xsd:element name="and" type="Binary" />
+            <xsd:element name="or" type="Binary" />
+            <xsd:element name="not" type="Unary" />
+            <xsd:element name="config">
+                <xsd:complexType>
+                        <xsd:attribute name="var" type="xsd:string" use="required"/>
+                </xsd:complexType>
+            </xsd:element>
+        </xsd:choice>
+    </xsd:group>
+
+  <xsd:complexType name="Unary">
+        <xsd:group ref="ConfigGroup"/>
+  </xsd:complexType>
+
+  <xsd:complexType name="Binary">
+    <xsd:sequence>
+            <xsd:group ref="ConfigGroup" minOccurs="2" maxOccurs="2"/>
```

Comment:
Is it really necessary to limit `maxOccurs` to 2?

Reply:
Probably not, but there wasn't anything that used more than 2.
I'm usually pretty conservative about adding things that aren't in use (Note I added `<or>` though despite nothing using it).
I can definitely willing to change it though if that is the consensus.

Reply:
So, I went ahead and did this change as it may make the (currently non-existent rust output) easier so you don't have to encode and as `all(a, all(b, c))` in theory.

Thank you for the review Axel, I believe i've covered all of your points now.